node/txnSyncConn.go

```diff
@@ -199,19 +201,15 @@ func (tsnc *transcationSyncNodeConnector) Handle(raw network.IncomingMessage) ne
 }
 
 func (tsnc *transcationSyncNodeConnector) stop() {
-
+	tsnc.txHandler.Stop()
 }
 
 func (tsnc *transcationSyncNodeConnector) IncomingTransactionGroups(networkPeer interface{}, txGroups []transactions.SignedTxGroup) (transactionPoolSize int) {
-	defer func() {
-		transactionPoolSize = tsnc.node.transactionPool.PendingCount()
-	}()
+	// count the transactions that we are adding.
+	txCount := 0
 	for _, txGroup := range txGroups {
-		if err := tsnc.txHandler.Handle(txGroup.Transactions); err != nil {
-			// we had some failuire, disconnect from peer.
-			tsnc.node.net.Disconnect(networkPeer)
-			break
-		}
+		txCount += len(txGroup.Transactions)
 	}
-	return
+	tsnc.txHandler.HandleTransactionGroups(networkPeer, txGroups)
```

Comment:
I think that you've got the gist here. Because we want to make it a background operation, we can no longer know what the size of the transaction pool exactly. So, we make a "best efforts attempt".
The returned number here doesn't have to be precise. It only helps us to determine if we need to adjust the beta, which get modified only on really large "chunks", so the txCount might not be as critical anyway.